src/api/actions/uploadInBackground.ts

```diff
@@ -0,0 +1,147 @@
+import { checkBulkUploadJob } from "./checkBulkUploadJob";
+import { startBulkUpload } from "./startBulkUpload";
+import {
+  JobIdToUrl,
+  UploadJobStatus,
+  UrlToIPFS,
+  UrlToJobId,
+} from "../../types";
+import * as fs from "fs";
+
+const URLS_PER_CHUNK_START = 100;
+const URLS_PER_CHUNK_CHECK = 1000;
+const ERROR_FILEPATH = "error.json";
```

Comment:
SDK functions shouldn't write to file. This makes the SDK inflexible on where/how it can be used. For instance if we're using the SDK in a browser, we can't be writing to disk.

Instead of writing to file, ask the user for a callback function which you will then pass to them any errors or failures.
Or, you could make the return type different, instead return an interface which has 3 fields "success" "failures" "errors"